consensus/hotstuff/votecollector/statemachine.go

```diff
@@ -119,6 +120,10 @@ func (m *VoteCollector) processVote(vote *model.Vote) error {
 				m.notifier.OnInvalidVoteDetected(vote)
 				return nil
 			}
+			if engine.IsDuplicatedEntryError(err) {
```

Comment:
Duplicated entry wasn't previously handled, this happens quite often in real world scenario. For know it's probably ok just to ignore it. Currently we don't check if the signature is same or not and it's crucial to do that if we want to slash.

Reply:
### ⚠️ ❗ 

Based on my understanding, this condition should never happen; i.e. it happening indices an unhandled edge case in our deduplication logic (or a misunderstanding on my side 😅 ). This is my understanding: 
1. before votes are processed in any form, they go to the cache (`Add` is an atomic operation): https://github.com/onflow/flow-go/blob/e4b98937e1c7926ff69a53366b4df1d82517db65/consensus/hotstuff/votecollector/statemachine.go#L90-L90
2. we already received a vote from the same node: https://github.com/onflow/flow-go/blob/e4b98937e1c7926ff69a53366b4df1d82517db65/consensus/hotstuff/votecollector/vote_cache.go#L74 and if it is for the same block, the cache returns a `RepeatedVoteErr` error: https://github.com/onflow/flow-go/blob/e4b98937e1c7926ff69a53366b4df1d82517db65/consensus/hotstuff/votecollector/vote_cache.go#L79
3. and we drop the vote: https://github.com/onflow/flow-go/blob/e4b98937e1c7926ff69a53366b4df1d82517db65/consensus/hotstuff/votecollector/statemachine.go#L92-L94

Based on this reasoning, it should be conceptually _impossible_ for a duplicated vote to reach the `VoteProcessor`. Even votes from byzantine nodes with different (invalid) signatures should not reach this point, as we deduplicate solely based on the voter's nodeID.